src/vm/database/clarity_db.rs

```diff
@@ -125,46 +150,108 @@ impl HeadersDB for &dyn HeadersDB {
     fn get_burn_block_height_for_block(&self, bhh: &StacksBlockId) -> Option<u32> {
         (*self).get_burn_block_height_for_block(bhh)
     }
-    fn get_miner_address(&self, bhh: &StacksBlockId)  -> Option<StacksAddress> {
+    fn get_miner_address(&self, bhh: &StacksBlockId) -> Option<StacksAddress> {
         (*self).get_miner_address(bhh)
     }
+    fn get_total_liquid_ustx(&self, bhh: &StacksBlockId) -> u128 {
+        (*self).get_total_liquid_ustx(bhh)
+    }
+}
+
+impl BurnStateDB for SortitionDBConn<'_> {
```

Comment:
I think the functions `get_burn_block_height` and `get_burn_header_hash` should be provided with a `SortitionId` (or, equivalently, the `ConsensusHash`) of the BlockSnapshot that selected the current Stacks block's parent, and _not_ invoke `inner_get_burn_block_height`/`inner_get_burn_header_hash`, but rather use that `SortitionId` to: 

* in the case of `get_burn_block_height`, just fetch that block snapshot and return the height
* in the case of `get_burn_header_hash`, call `get_block_snapshot_by_height` and then return that snapshot's burn header hash.

The reason for this is that, as implemented here, if the SortitionDB is far ahead of the stacks chain state, then `get_burn_block_height` could return a block height _greater_ than the one that selected the current Stacks block (also, it could return information for a block that may not exist in the current Bitcoin fork). And `get_burn_header_hash` could have the same problem (returning the burn header for a block with height greater than the current block).

I can't tell from looking at the code where these functions get used currently, but if we want to eventually expose the burnchain lookup functionality to Clarity, these functions would need to be evaluated in the context of the current Stacks chain tip's associated BlockSnapshot.

Reply:
Yes; I will make this happen.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1850/commits/ba7145851dac352e801dd1b4b3b5eff1fe62093d